clients/input_copy.go

```diff
@@ -213,6 +274,25 @@ func CopyFile(ctx context.Context, sourceURL, destOSBaseURL, filename, requestID
 	return
 }
 
+func CopyReaderFile(ctx context.Context, readerFile io.Reader, destOSBaseURL, filename, requestID string) (writtenBytes int64, err error) {
+	err = backoff.Retry(func() error {
```

Comment:
This `retry` will not work since you're using the same reader/stream instance. It's possible that it fails the first time and then it only uploads a partial file on the next attempt.

So you must have the download+decryption logic inside the retry, which is why I suggested you pass in a lambda func to handle it transparently for the copying logic here.